AK/Buffered.h

```diff
@@ -57,15 +57,13 @@ class Buffered<StreamType, Size, typename EnableIf<IsBaseOf<InputStream, StreamT
         auto nread = buffer().trim(m_buffered).copy_trimmed_to(bytes);
 
         m_buffered -= nread;
-        buffer().slice(nread, m_buffered).copy_to(buffer());
+        if (m_buffered > 0)
+            buffer().slice(nread, m_buffered).copy_to(buffer());
 
         if (nread < bytes.size()) {
-            m_buffered = m_stream.read(buffer());
+            nread += m_stream.read(bytes.slice(nread));
```

Comment:
Again mentioning the actual data for perf impact here in the commit message would be nice.